TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift

```diff
@@ -318,14 +336,28 @@ import PromiseKit
                                             userToken: String?,
                                             broker: String,
                                             onSuccess: @escaping (_ linkedBroker: TradeItLinkedBroker) -> Void,
+                                            onSecurityQuestion: @escaping (TradeItSecurityQuestionResult,
+                                                _ submitAnswer: @escaping (String) -> Void,
+                                                _ onCancelSecurityQuestion: @escaping () -> Void
+                                            ) -> Void,
                                             onFailure: @escaping (TradeItErrorResult) -> Void) {
         let linkedLogin = self.connector.saveToKeychain(withUserId: userId, andUserToken: userToken, andBroker: broker, andLabel: broker)
 
         if let linkedLogin = linkedLogin, let userId = userId, let userToken = userToken {
             let linkedBroker = self.loadLinkedBrokerFromLinkedLogin(linkedLogin)
-            self.linkedBrokers.append(linkedBroker)
-            self.oAuthDelegate?.didLink?(linkedBroker: linkedBroker, userId: userId, userToken: userToken)
-            onSuccess(linkedBroker)
+
+            linkedBroker.authenticateIfNeeded(
+                onSuccess: {
+                    self.linkedBrokers.append(linkedBroker)
+                    self.oAuthDelegate?.didLink?(linkedBroker: linkedBroker, userId: userId, userToken: userToken)
+                    onSuccess(linkedBroker)
+                },
+                onSecurityQuestion: onSecurityQuestion,
+                onFailure: { error in
```

Comment:
Change `onFailure: { error in ... }` to `onFailure: onSuccess` because a failed authentication doesn't mean the linking failed.